### ISSUE TITLE
how to use a custom text _id

### ISSUE THREAD

**Issue by gfanini** [10gen-labs/mongo-php-driver-prototype#1264]
how could I use a custom text _id like "0F000000" with the library, with php 7
since it is allowed by mongodb : https://docs.mongodb.com/manual/core/index-text/

```
$recordid='0F000000';
$uri = 'mongodb://'.$dbhost.$g_replicaSet;		
$client=new MongoDB\Client($uri);		
$collection = $client->test->test_collection;  	
$item = $collection->findOne(array('_id' => new MongoDB\BSON\ObjectId($recordid)));
```

this gives an exception, I tried with regex but it unexplainably does not fetch any records from some _id, how could I debug why
the cursor comes back empty with 

```
$regex = new MongoDB\BSON\Regex ('0F000000', 'i');
$query = new MongoDB\Driver\Query( array('_id' => $regex) );
```

the same regex works in mongo shell, perhaps enabling debug trace in mongo php ? I am launching from php 7.4 and 8 cli, not from apache say

**Comment by gfanini** [comment]
do you have perhaps any clue what could cause these same types of queries above to return nothing in linux debian 9 stretch 

# locate mongodb.so
/usr/lib/php/20190902/mongodb.so
/usr/lib/php/20200930/mongodb.so
# ls -la /usr/lib/php/20190902/mongodb.so
-rw-r--r-- 1 root root 1516736 Aug  2 09:20 /usr/lib/php/20190902/mongodb.so
# ls -la /usr/lib/php/20200930/mongodb.so
-rw-r--r-- 1 root root 1516736 Aug  2 09:20 /usr/lib/php/20200930/mongodb.so
# pecl install mongodb
PHP Warning:  Module 'mongodb' already loaded in Unknown on line 0
pecl/mongodb is already installed and is the same as the released version 1.10.0
install failed

whereas it works reliably without a glitch in windows 10 

C:\wamp64\bin\php\php7.2.18\ext

05/09/2017  16:57           348.672 php_mongo.dll
09/09/2019  17:31           918.528 php_mongodb.dll

**Comment by jmikola** [comment]
> do you have perhaps any clue what could cause these same types of queries above to return nothing in linux debian 9 stretch

Nothing in the script I shared is specific to an operating system, or even a recent version of the extension/library.

Based on the "Module 'mongodb' already loaded" warnings in the output you shared, you likely have some redundant lines in your PHP configuration (either `php.ini` or your within your scan directory for additional INIs); however, I don't imagine that would prevent the script from inserting and querying a document.

I would start by checking if the queries are actually executing successfully and returning a cursor with no results (e.g. `var_dump()` prints an empty array), or if there is some exception occurring that you're not seeing. Beyond that, try simplifying the script to query without a filter and see if you can at least return the two documents that were inserted.
